diff --git a/drivers/mmc/host/pxamci.c b/drivers/mmc/host/pxamci.c
index 38dfbf6..4f74cef 100644
--- a/drivers/mmc/host/pxamci.c
+++ b/drivers/mmc/host/pxamci.c
@@ -835,6 +835,9 @@ static int pxamci_probe(struct platform_device *pdev)
 			0 : MMC_CAP2_RO_ACTIVE_HIGH;
 	}
 
+	/* Set active high Card Detect for Zipit Z2 */
+	mmc->caps2 |= MMC_CAP2_CD_ACTIVE_HIGH;
+
 	if (gpio_is_valid(gpio_cd))
 		ret = mmc_gpio_request_cd(mmc, gpio_cd, 0);
 	if (ret) {
